x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_editor.tsx

```diff
@@ -333,6 +334,16 @@ export function DimensionEditor(props: DimensionEditorProps) {
           </EuiFormRow>
         ) : null}
 
+        {!currentFieldIsInvalid && !incompatibleSelectedOperationType && selectedColumn && (
+          <TimeScaling
+            selectedColumn={selectedColumn}
+            columnId={columnId}
+            layerId={layerId}
+            state={state}
+            setState={setState}
```

Comment:
I have been testing this out for the reference-based UI. How about changing the arguments here to pass `layer` and a new function, instead of requiring the UI to update the entire state: `updateLayer={(newLayer: IndexPatternLayer) => setState(mergeLayer({ layerId, indexPattern, newLayer }))}`